libcxx/include/__tuple/ignore.h

```diff
@@ -0,0 +1,35 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCPP___TUPLE_IGNORE_H
+#define _LIBCPP___TUPLE_IGNORE_H
+
+#include <__config>
+
+#if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
+#  pragma GCC system_header
+#endif
+
+#ifndef _LIBCPP_CXX03_LANG
+
+_LIBCPP_BEGIN_NAMESPACE_STD
+
+struct __ignore_type {
+  template <class _Tp>
+  _LIBCPP_HIDE_FROM_ABI constexpr const __ignore_type& operator=(const _Tp&) const noexcept {
+    return *this;
+  }
+};
+
+inline constexpr __ignore_type ignore;
```

Comment:
This change seems like an ABI break, can you add `_LIBCPP_HIDE_FROM_ABI`?

Reply:
I'm not sure what ABI break you see, but this definitely shouldn't be makred `_LIBCPP_HIDE_FROM_ABI`, since that would be non-conforming. You would get different addresses in different dylibs.

Reply:
The type of the `ignore` changes and that type is not identical. Why would adding `_LIBCPP_HIDE_FROM_ABI` be non-conforming?

Reply:
Sure it's technically an ODR violation, but that's benign AFAICT. We don't change the layout in any way. As I already said, it's non-conforming because the object would have different addresses in different dylibs.

Reply:
@ldionne are you happy with this change without using `_LIBCPP_HIDE_FROM_ABI`. 

Basically the change is a a different type and the signature `operator=(_Tp&&) const` changed to `operator=(const _Tp&) const noexcept`

Reply:
The current way it is defined seems correct to me. We do need to ensure that `ignore` has the same address across all TUs since the Standard says `inline constexpr` explicitly, so we can't ABI-tag this.

I do agree this is technically an ABI break due to the change from `__ignore_t<unsigned char>` to `__ignore_type`, however I really really don't think it matters in practice. I don't even feel it is worth release-noting this ABI change since I think essentially nobody would ever notice.